ui/pages/settings/advanced-tab/advanced-tab.component.js

```diff
@@ -117,6 +121,113 @@ export default class AdvancedTab extends PureComponent {
     );
   }
 
+  async getTextFromFile(file) {
+    return new Promise((resolve, reject) => {
+      const reader = new window.FileReader();
+      reader.onload = (e) => {
+        const text = e.target.result;
+        resolve(text);
+      };
+
+      reader.onerror = (e) => {
+        reject(e);
+      };
+
+      reader.readAsText(file);
+    });
+  }
+
+  async handleFileUpload(event) {
+    const file = event.target.files[0];
+    const jsonString = await this.getTextFromFile(file);
```

Comment:
Add validation here to address Mariona's 3rd comment?

Reply:
actually, Olu and I discussed a better solution: we just won't backup identies and selected address data